### PR TITLE
chore(weave): less confusing log configuration

### PR DESCRIPTION
**Pull request by laxels**
See individual comments for details.

### CODE REVIEW

File: weave/logs.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ class LogFormat(enum.Enum):
 @dataclasses.dataclass
 class LogSettings:
     format: LogFormat
-    level: str
+    level: typing.Optional[int]
```
Comment (laxels) [inline weave/logs.py:29]: To eliminate the need for converting back/forth between `str` and `int` log levels, unify everything to `int`.

`None` here is allowed as a "just follow the logger's level" option.

File: weave/logs.py [hunk 2]
```diff
@@ -102,13 +102,18 @@ def default_log_filename() -> typing.Optional[str]:
...
+            return level_int_from_env
+
     if os.environ.get("WEAVE_SERVER_ENABLE_LOGGING"):
-        default_level = "INFO"
-    level = os.environ.get("WEAVE_LOG_LEVEL", default_level)
-    return logging.getLevelName(level)
+        return logging.INFO
+
+    # Default to only showing ERROR logs
+    return logging.ERROR
```
Comment (laxels) [inline weave/logs.py:116]: - The only time we ever need to convert from a string log level is when reading `WEAVE_LOG_LEVEL` env var
- Branching logic reorganized for additional intuitiveness

File: weave/logs.py [hunk 3]
```diff
@@ -227,8 +232,8 @@ def configure_logger() -> None:
             # terminal, you get the logs.
             enable_stream_logging(
                 logger,
-                wsgi_stream_settings=LogSettings(LogFormat.PRETTY, "INFO"),
-                pid_logfile_settings=LogSettings(LogFormat.PRETTY, "INFO"),
+                wsgi_stream_settings=LogSettings(LogFormat.PRETTY, level=None),
```
Comment (laxels) [inline weave/logs.py:235]: The explicit `logging.INFO` level here meant that debug logs would not show in the dev server even when `WEAVE_LOG_LEVEL=DEBUG`.

File: weave/logs.py [hunk 4]
```diff
@@ -237,28 +242,28 @@ def configure_logger() -> None:
...
         else:
             # Otherwise this is dev mode with datadog logging turned on.
             # Log to standard out and a known filename that the datadog
             # agent can watch.
             enable_stream_logging(
                 logger,
-                wsgi_stream_settings=LogSettings(LogFormat.PRETTY, "DEBUG"),
-                server_logfile_settings=LogSettings(LogFormat.DATADOG, "DEBUG"),
+                wsgi_stream_settings=LogSettings(LogFormat.PRETTY, level=None),
+                server_logfile_settings=LogSettings(LogFormat.DATADOG, level=None),
```
Comment (laxels) [inline weave/logs.py:268]: Since `DEBUG` is the most permissive level, it's essentially the same as the handler just following the logger's level.

**Review by shawnlewis — APPROVED** [review]

**Pull request merged**